Re-add inbuild signing

mmitche opened the pull request:
This re-enables inbuild signing. Unlike the previous iteration of inbuild signing we now can use the same approach for dealing with containers as we do with post-build signing (sign wixpacks, repack them, iteratively). Essentially a way to think of this is that we are doing post-build signing, but it happens to be in-build.

To do so, a target is added in the prepare-artifacts project that calls the arcade sign project (windowsdesktop doesn't really use the standard arcade process).

mmitche commented:
Inbuild test: https://dev.azure.com/dnceng/internal/_build/results?buildId=1891306&view=results
Post-build (default): https://dev.azure.com/dnceng/internal/_build/results?buildId=1891301&view=results

hoyosjs reviewed (approved):
Looks sane to me if the builds pass and all the assets end up signed in-build.

jkoritzinsky reviewed (approved):
Changes look good as long as it works.

mmitche commented:
It dropped back to MIcrosoft400 due to some quirkiness in how this repo does its publishing/signing. Fixed I think, updated builds:

In-build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1891485&view=results
Post-build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1891486&view=results

hoyosjs commented:
@mmitche metadata looks good with `MicrosoftDotNet500` being used instead of 400.

RussKie reviewed (commented):
👍
